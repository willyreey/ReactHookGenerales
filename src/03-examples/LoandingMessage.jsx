export const LoandingMessage = () => {
  return (
    <section 
        style={{ heigth: 200 }}
        className="alert alert-info text-center p-0 m-0 d-flex align-item-center justify-content-center"
    >
        <h1>Cargando...</h1>

    </section>
  )
}
