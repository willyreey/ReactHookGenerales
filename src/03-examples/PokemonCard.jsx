import { useRef } from "react";
import { useLayoutEffect } from "react"

export const PokemonCard = ({id, name, sprites = []}) => {

  const pRef = useRef()

  useLayoutEffect(() => {
    console.log(pRef.current.getBoundingClientRect());

  }, [name])

  return (
    <section 
    style={{ display: 'flex' }}
    ref={pRef}
    >  
    <h2>#{id} - {name}</h2>

    <div>
        {
            sprites.map( sprite => (
                <img key={sprite} src={sprite} alt={name} />
            ))
        }
    </div>

</section>
  )
}
