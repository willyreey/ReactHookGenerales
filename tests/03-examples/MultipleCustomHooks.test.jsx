import { fireEvent, render, screen } from "@testing-library/react"
import { MultipleCustomHooks } from "../../src/03-examples/MultipleCustomHooks"
import { useCounter } from "../../src/hooks/useCounter"
import { useFetch } from "../../src/hooks/useFetch"

jest.mock('../../src/hooks/useFetch')
jest.mock('../../src/hooks/useCounter')


describe('Pruebas en MultipleCustomHooks', () => { 

    const mockIncrement = jest.fn()
    useCounter.mockReturnValue({
        counter: 1,
        increment: mockIncrement
    })

    beforeEach(() => {
        jest.clearAllMocks()
    })

    test('Debe de mostrar el componente por defecto', () => { 

        useFetch.mockReturnValue({
            data: null, 
            isLoanding: true, 
            hasError: null
        })

        render(<MultipleCustomHooks/>)

        expect(screen.getByText('Cargando...'))
        expect(screen.getByText('Informacion del Pokemon'))
        expect().toMatchSnapshot()
        
     })

    test('Debe de mostrar un Pokemon', () => { 

        useFetch.mockReturnValue({
            data: {id: 1, name: 'Picachu',sprites: {front_default:'htpps://hola.com',front_shiny:'htpps://hola.com',back_default:'htpps://hola.com'}}, 
            isLoanding: false, 
            hasError: null
        })

        render(<MultipleCustomHooks/>)
        expect( screen.getByText('Informacion del Pokemon') ).toBeTruthy()
        expect( screen.getByRole('button', {name:'Anterior'}) ).toBeTruthy()
        
    })

    test('Debe de llamar la funcion de incrementar', () => { 
        
        useFetch.mockReturnValue({
            data: {id: 1, name: 'Picachu',sprites: {front_default:'htpps://hola.com',front_shiny:'htpps://hola.com',back_default:'htpps://hola.com'}}, 
            isLoanding: false, 
            hasError: null
        })

        render(<MultipleCustomHooks/>)
        const nextButton = screen.getByRole('button', {name:'Siguiente'})
        fireEvent.click(nextButton)
        screen.debug()
        expect(mockIncrement).toHaveBeenCalled();
        
    })

 })