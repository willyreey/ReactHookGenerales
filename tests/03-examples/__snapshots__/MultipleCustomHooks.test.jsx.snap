// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultipleCustomHooks Debe de mostrar el componente por defecto 1`] = `undefined`;
